App/Redux/TextileEventsRedux.ts

```diff
@@ -1,42 +1,34 @@
 import { createAction, getType, ActionType } from 'typesafe-actions'
-import { NodeState } from '@textile/react-native-sdk'
 import { RootState } from './Types'
 
+export enum NodeState {
+  stopped = 'stopped',
+  started = 'started',
+  online = 'online'
+}
+
 const actions = {
-  newNodeState: createAction('@bridge/NEW_NODE_STATE', (resolve) => {
-    return (state: NodeState) => resolve({ state })
+  nodeStarted: createAction('NODE_STARTED'),
+  nodeStopped: createAction('NODE_STOPPED'),
+  nodeOnline: createAction('NODE_ONLINE'),
+  nodeFailedToStart: createAction('NODE_FAILED_TO_START', (resolve) => {
+    return (error: string) => resolve({ error })
+  }),
+  nodeFailedToStop: createAction('NODE_FAILED_TO_STOP', (resolve) => {
+    return (error: string) => resolve({ error })
   }),
-  appStateChange: createAction(
-    '@bridge/APP_STATE_CHANGE',
-    (resolve) => (previousState: string, newState: string) => resolve({ previousState, newState })
-  ),
-  newErrorMessage: createAction(
-    '@bridge/NEW_ERROR_MESSAGE',
-    (resolve) => (type: string, message: string) => resolve({ type, message })
-  ),
-  startNodeFinished: createAction(
-    '@bridge/startNodeFinished',
-    (resolve) => () => resolve()
-  ),
   stopNodeAfterDelayStarting: createAction(
     '@bridge/stopNodeAfterDelayStarting',
-    (resolve) => () => resolve()
+    (resolve) => (delay: number) => resolve({ delay })
```

Comment:
The callback for running the node in the background provides the amount of time it will run for, which is dynamically calculated on iOS.